unit_tests/schema_tests.json

```diff
@@ -0,0 +1,579 @@
+          "description": "cannot have spaces"
+        }
+      ]
+    },
+    "title_number":{
+      "pattern_path": "/definitions/title_number",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "DN1234",
+          "expected": true,
+          "description": "valid title number"
+        },
+        {
+          "payload": "12345",
+          "expected": true,
+          "description": "option start characters must start with a 1"
+        },
+        {
+          "payload": "022222",
+          "expected": false,
+          "description": "must start with a 1-9 after initial chars"
+        },
+        {
+          "payload": "DN02222",
+          "expected": false,
+          "description": "must start with a 1-9 after initial chars"
+        },
+        {
+          "payload": "DN02222",
+          "expected": false,
+          "description": "must start with a 1-9 after initial chars"
+        },
+        {
+          "payload": "1234Z",
+          "expected": true,
+          "description": "can end with Z"
+        },
+        {
+          "payload": "1234T",
+          "expected": true,
+          "description": "can end with t"
+        },
+        {
+          "payload": "DN1234T",
+          "expected": false,
+          "description": "can end with t only when no starting alphas"
+        }
+      ]
+    }
+  },
+  "test_payloads": [
+    { "payload": {
+         "title_number": "DT567568",
+          "md_ref":"e-MD12345",
+         "borrowers": [
+            {
+                "forename": "Amanda",
+                "middle_name": "Elizabeth",
+                "surname": "Smythe",
+                "gender": "Female",
+                "address": "1 The High Street Highley CR0 9YH",
+                "dob": "21/09/1976",
+                "phone_number": "07507154077"
+            },
+            {
+                "forename": "Paul",
+                "middle_name": "",
+                "surname": "Smythe",
+                "gender": "Not Specified",
+                "address": "1 The High Street Highley PL6 7TG",
+                "dob": "21/08/2015",
+                "phone_number": "07507154057"
+            },
+            {
+                "forename": "Paul",
+                "middle_name": "",
+                "surname": "Smythe",
+                "gender": "Male",
+                "address": "1 The High Street Highley PL6 7TG",
+                "dob": "21/08/2015",
+                "phone_number": "07507154078"
+            }
+
+         ]
+         }, "description": "valid payload with multiple borrowers", "expected": 0
+    },
+    { "payload": {
+         "title_number": "DT567568",
+          "md_ref":"e-MD12345",
+         "borrowers": [
+            {
+                "forename": "Amanda",
+                "middle_name": "Elizabeth",
+                "surname": "Smy the",
```

Comment:
I'm not 100% sure on this but i have known people with non hyphenated double barrel names might be worth confirming the business rule on this